mycli build failes on automated tests

### Steps To Reproduce
Steps to reproduce the behavior:
1. build mycli on unstable or master branch

### Build log
```
mycli> =================================== FAILURES ===================================
mycli> _________________________ test_auto_escaped_col_names __________________________
mycli> 
mycli> completer = <mycli.sqlcompleter.SQLCompleter object at 0x7fffe752ecb0>
mycli> complete_event = <Mock id='140737074361776'>
mycli> 
mycli>     def test_auto_escaped_col_names(completer, complete_event):
mycli>         text = 'SELECT  from `select`'
mycli>         position = len('SELECT ')
mycli>         result = list(completer.get_completions(
mycli>             Document(text=text, cursor_position=position),
mycli>             complete_event))
mycli> >       assert result == [
mycli>             Completion(text='*', start_position=0),
mycli>             Completion(text='`ABC`', start_position=0),
mycli>             Completion(text='`insert`', start_position=0),
mycli>             Completion(text='id', start_position=0),
mycli>         ] + \
mycli>             list(map(Completion, completer.functions)) + \
mycli>             [Completion(text='`select`', start_position=0)] + \
mycli>             list(map(Completion, completer.keywords))
mycli> E       AssertionError: assert [Completion(t...CAT')])), ...] == [Completion(t...CAT')])), ...]
mycli> E         At index 23 diff: Completion(text='select', start_position=0, display=FormattedText([('', 'select')])) != Completion(text='`select`', start_position=0, display=FormattedText([('', '`select`')]))
mycli> E         Use -v to get more diff
mycli> 
mycli> complete_event = <Mock id='140737074361776'>
mycli> completer  = <mycli.sqlcompleter.SQLCompleter object at 0x7fffe752ecb0>
mycli> position   = 7
mycli> result     = [Completion(text='*', start_position=0, display=FormattedText([('', '*')])), Completion(text='`ABC`', start_position=0...rmattedText([('', 'AVG')])), Completion(text='CONCAT', start_position=0, display=FormattedText([('', 'CONCAT')])), ...]
mycli> text       = 'SELECT  from `select`'
mycli> 
mycli> test/test_smart_completion_public_schema_only.py:318: AssertionError
mycli> =========================== short test summary info ============================
mycli> FAILED test/test_smart_completion_public_schema_only.py::test_auto_escaped_col_names - AssertionError: assert [Completion(t...CAT')])), ...] == [Completion(t...CA...
mycli> ======= 1 failed, 232 passed, 41 skipped, 1 xfailed, 1 xpassed in 4.84s ========
```

full log: https://gist.github.com/dev-null-undefined/8a7eb48188d0403cdef7d114cad52b89


### Notify maintainers
@jojosch 

### Metadata
```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.3, NixOS, 23.05 (Stoat), 23.05.20230113.befc839`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/ci0324339l5i3x7q68vigpfcf1jicsgs-source`
```

--- a/pkgs/tools/admin/mycli/default.nix
+++ b/pkgs/tools/admin/mycli/default.nix
@@ -30,15 +30,23 @@ buildPythonApplication rec {
     sqlparse
   ];
 
-  nativeCheckInputs = [ pytest glibcLocales ];
+  nativeCheckInputs = [ pytestCheckHook glibcLocales ];
 
-  checkPhase = ''
+  preCheck = ''
     export HOME=.
     export LC_ALL="en_US.UTF-8"
-
-    py.test \
-      --ignore=mycli/packages/paramiko_stub/__init__.py
   '';
+
+  disabledTestPaths = [
+    "mycli/packages/paramiko_stub/__init__.py"
+  ];
+
+  disabledTests = [
+    # Note: test_auto_escaped_col_names is currently failing due to a bug upstream.
+    # TODO: re-enable this test once there is a fix upstream. See
+    # https://github.com/dbcli/mycli/issues/1103 for details.
+    "test_auto_escaped_col_names"
+  ];
 
   postPatch = ''
     substituteInPlace setup.py \

